lib/facility_factory.rb

```diff
@@ -0,0 +1,55 @@
+class FacilityFactory
+
+    def create_facilities(locations)
+        if locations[0][:state] == "CO"
+            create_co_facilities(locations)
+        elsif locations[0][:state] == "NY"
+            create_ny_facilities(locations)
+        elsif locations[0][:state] == "MO"
+            create_mo_facilities(locations)
+        end
+    end
```

Comment:
More helper methods! I like that! Makes it very easy to read the control flow of the method.